src/nbl/video/COpenGLCommandBuffer.cpp

```diff
@@ -96,6 +101,18 @@ namespace video
         }
     }
 
+    bool COpenGLCommandBuffer::reset(uint32_t _flags)
+    {
+        if (!(m_cmdpool->getCreationFlags() & IGPUCommandPool::ECF_RESET_COMMAND_BUFFER_BIT))
+            return false;
+
+        freeSpaceInCmdPool();
+        m_commands.clear();
+        IGPUCommandBuffer::reset(_flags);
+
+        return true;
+    }
```

Comment:
you need an atomic counter here, the command buffers you allocate need to cache the value of this counter when constructing and at every `begin()` call, if the value changes it means a reset happened that affects the command buffer (you can drop its state and update the cached value).

Now obviously if the value changed and the state of the commandbuffer is wrong (you are sure its actually PENDING or INVALID) then you throw a big error.